sale_import_delivery_carrier/components/importer_sale_channel.py

```diff
@@ -0,0 +1,40 @@
+from odoo.addons.component.core import Component
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "importer.sale.channel"
+
+    def _prepare_vals(self, data):
+        vals = super()._prepare_vals(data)
+        return self._add_delivery_line(vals, data)
+
+    def _add_delivery_line(self, vals, data):
+        if not data.get("delivery_carrier"):
+            return data
+        delivery_carrier = self.env["delivery.carrier"].search(
+            [("name", "=", data["delivery_carrier"]["name"])]
+        )
+        delivery_line = dict()
+        delivery_line["carrier_id"] = delivery_carrier.id
+        price = data["delivery_carrier"]["price_unit"]
+        discount = data["delivery_carrier"]["discount"]
+        partner = self.env["res.partner"].browse(vals["partner_id"])
+        carrier_with_partner_lang = delivery_carrier.with_context(lang=partner.lang)
+        if carrier_with_partner_lang.product_id.description_sale:
+            description = "{}: {}".format(
+                carrier_with_partner_lang.name,
+                carrier_with_partner_lang.product_id.description_sale,
+            )
+        else:
+            description = carrier_with_partner_lang.name
+        new_line_vals = {
+            "name": description,
+            "product_uom_qty": 1,
+            "product_uom": delivery_carrier.product_id.uom_id.id,
+            "product_id": delivery_carrier.product_id.id,
+            "price_unit": price,
+            "discount": discount,
+            "is_delivery": True,
```

Comment:
For our imported delivery line, we don't care for fiscal position or taxes. Is this right for our use case ?

Reply:
fiscal position and tax must be correct. Create the test and then we will see how to implement it